documentation/proposals/Proposal - Multi-Backend Input.md

```diff
@@ -834,6 +831,103 @@ public interface IJoystickInputHandler : IInputHandler
 
 `HandleHatMove` must be called when any value of `JoystickState.Hats` changes.
 
+# Input Maps
+
+## Presentation
+Input maps are designed to be easyer to the user to handle actions with more than one input.
+At first view it can looks useless and a bad practice, but this is complemtelly useful for cross-plataform programs, being games an example.
+
+This is an pseudocode example of what the Input Map system solves:
+
+Without inputMaps:
+```cs
+inputContext.Update();
+
+private Dictionary<string, bool> pressedKeys = new();
+
+inputContext.Gamepads.ThumbstickMove += @event => {Move(@event.Value)};
+inputContext.Joysticks.AxisMove += @event => {Move(@event.Value)};
+
+private Move(Vector2D<float> value)
+{
+    // Code for movement here
+}
+```
+
+With inputMaps:
+```cs
+inputContext.Update();
+
+// This single line get rid of the two in the example above. TWO ISN'T THE LIMIT!
+inputContext.Map.GetAction("Move").Axis += @event => {Move(@event.Value)};
+
+private Move(Vector2D<float> value)
+{
+    // Code for movement here
+}
+```
+
+## Implementation
+Each InputContext have one reference to an InputMap.
```

Comment:
Why have you done the ownership this way around? Surely it makes sense for an input map to own an input context and not the other way around given that the map is essentially an extension/grouping of the existing functionality? This also relates to my other comment about making this a separate proposal.

Reply:
the map is an extension/grouping of the receiving of input events. It's on the same level of a ButtonDown or a ButtonUp, the only difference is because it will filtrate and operate the inputs before send it back.
i really can't see how a InputMap can own an InputContext because of it

Reply:
Nonetheless, we are eager to ensure that this is strictly an extension to the core API and not part of the core API itself, so consider other ways to express this relationship. Inheritance maybe? Composition?

Reply:
I really can't see how i can make it work with inheritance becuase the inputContext are really complex and there's a lot od useless information to the maps, so i will try to do composition, also because you will always get an InputContext from the core window so (i really don't want to make it using extension methods, i believe in composition supremacy)

Reply:
For reasons such as this, I really wish you had waited to make this proposal to be honest, this isn't a one person job by any stretch of the imagination and we're likely to have to back-track on anything decided here once we actually start implementing it, which could be years from now.

Feel free to come up with an alternative design, but I don't think we can have enough confidence having so little available information. Both the windowing and input proposals came from multiple maintainers experimenting with potential designs and actively collaborating in a short space of time to come up with what feels the most natural both to use and implement, and we just don't have this information available for your proposal because it is a one person job. That's not to say that one person proposals aren't accepted, they definitely have been in the past, but historically these have been the proposals we've come to regret implementing the most. And we're just not interested in doing that experimentation right now because we have so many more pressing priorities before we can even look at extensions such as this. This is also one of the many reasons I put a comment to make this a separate proposal, so that it doesn't impact our much higher priority work.

Reply:
I understand, i will put this section in a separated proposal and try to improve the maximum that i can

Reply:
Also you want to hold it for a while until someone help to develop it better? Posting the proposal looks like the best way people can see and modify it, but it will be a problem if it gets rejected

Reply:
I think placing it in a separate proposal and merging it with an `(On Hold)` prefix is the best course of action.